ksmt-yices/src/main/kotlin/org/ksmt/solver/yices/KYicesExprConverter.kt

```diff
@@ -0,0 +1,565 @@
+                expr.convert(yicesArgs) { x: KExpr<KArithSort<*>> ->
+                    val isReal = x.sort == realSort
+                    val condition = if (isReal)
+                        mkArithGe(x as KExpr<KRealSort>, mkRealNum(0))
+                    else
+                        mkArithGe(x as KExpr<KIntSort>, mkIntNum(0))
+
+                    mkIte(condition, x, mkArithUnaryMinus(x))
+                }
+            }
+            Constructor.CEIL -> {
+                expr.convert(yicesArgs) { x: KExpr<KArithSort<*>> ->
+                    if(x.sort == intSort)
+                        x
+                    else
+                        mkCeil(x as KExpr<KRealSort>)
+                }
+            }
+            Constructor.FLOOR -> {
+                expr.convert(yicesArgs) { x: KExpr<KArithSort<*>> ->
+                    if(x.sort == intSort)
+                        x
+                    else
+                        mkFloor(x as KExpr<KRealSort>)
+                }
+            }
+            Constructor.RDIV -> {
+                argTypes.addAll(List(numChildren) { realSort })
+                expr.convert<KArithSort<*>, KArithSort<*>, KArithSort<*>>(yicesArgs, ::mkArithDiv)
+            }
+            Constructor.IDIV -> {
+                argTypes.addAll(List(numChildren) { realSort })
+                expr.convert(yicesArgs, ::mkIDiv)
+            }
+            Constructor.IMOD -> {
+                argTypes.addAll(List(numChildren) { realSort })
+                expr.convert(yicesArgs) { arg0: KExpr<KRealSort>, arg1: KExpr<KRealSort> ->
+                    val div = mkArithDiv(arg0, arg1)
+                    val mul = mkArithDiv(arg1, div)
+
+                    mkArithSub(arg0, mul)
+                }
+            }
+            Constructor.IS_INT_ATOM -> {
+                argTypes.addAll(List(numChildren) { realSort })
+                expr.convert(yicesArgs, ::mkRealIsInt)
+            }
+            Constructor.DIVIDES_ATOM -> {
+                argTypes.addAll(List(numChildren) { realSort })
+                expr.convert(yicesArgs) { arg0: KExpr<KRealSort>, arg1: KExpr<KRealSort> ->
+                    val div = mkIDiv(arg0, arg1)
+                    val rem = mkArithSub(arg0, mkArithMul(mkIntToReal(div), arg1))
+
+                    mkEq(rem, mkRealNum(0))
+                }
+            }
+            Constructor.ARITH_ROOT_ATOM -> TODO("ARITH_ROOT conversion is not supported")
+            Constructor.TUPLE_TERM -> TODO("Tuple conversion is not supported")
+            Constructor.CONSTRUCTOR_ERROR -> error("Constructor error")
+            else -> error("Unexpected constructor ${Terms.constructor(expr)}")
+        }
+    }
+
+    private fun KSort.toArraySort(): KArraySort<*, *> {
+        check(this is KArraySort<*, *>) { "Unexpected sort $this" }
+
+        return this
+    }
+
+    private inner class KYicesExprPreprocessor {
+        private val processedExpr = mutableSetOf<YicesTerm>()
```

Comment:
`mutableSetOf == LinkedHashSet`. If you don't need to preserve insertion order, it's better to use `HashSet`.